foundation/plugins/org.jboss.tools.foundation.core/src/org/jboss/tools/foundation/core/JVMProblemDetector.java

```diff
@@ -0,0 +1,475 @@
+ * This program is made available under the terms of the 
+ * Eclipse Public License v1.0 which accompanies this distribution, 
+ * and is available at http://www.eclipse.org/legal/epl-v10.html 
+ * 
+ * Contributors: 
+ * Red Hat, Inc. - initial API and implementation 
+ ******************************************************************************/ 
+package org.jboss.tools.foundation.core;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.StringReader;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.LinkedBlockingDeque;
+
+import org.eclipse.core.runtime.ILogListener;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Platform;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.osgi.util.NLS;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.ui.IStartup;
+import org.eclipse.ui.PlatformUI;
+import org.eclipse.ui.internal.views.log.LogSession;
+import org.eclipse.ui.internal.views.log.TailInputStream;
+import org.eclipse.ui.progress.UIJob;
+
+
+public class JVMProblemDetector implements IStartup, ILogListener{
+	private static final String PREFERENCE_NAME="org.jboss.tools.foundation.core.JVMProblemDetector_NOT_TO_SHOW"; //$NON-NLS-1$
+	private static final String STRING_1 = "org.osgi.framework.BundleException: Could not resolve module:"; //$NON-NLS-1$
+	private static final String STRING_2 = "-> Bundle-SymbolicName:"; //$NON-NLS-1$
+	private static final String STRING_3 = "Unresolved requirement: Require-Capability: osgi.ee; filter:=\"(&(osgi.ee="; //$NON-NLS-1$
+	private static final String STRING_4 = ")(version="; //$NON-NLS-1$
+	
+	private static final long MAX_LENGTH = 1024 * 1024;
+	
+	private UnresolvedStructure structure = new UnresolvedStructure();
+	
+	private LinkedBlockingDeque<String> queue = new LinkedBlockingDeque<String>();
+	
+	private String javaVersion = null;
+	
+	@Override
+	public void earlyStartup() {
+		PlatformUI.getPreferenceStore().setValue(PREFERENCE_NAME, false);
+		if(isAllowedToShow()){
+			javaVersion = System.getProperty("java.version"); //$NON-NLS-1$
+			
+			// listen log
+			Platform.addLogListener(this);
+			
+			// read error log
+			readLogFile();
+			
+			// read queue
+			readQueue();
+		}
+	}
+	
+	private void readLogFile() {
+		File logFile = Platform.getLogFileLocation().toFile();
+		BufferedReader reader = null;
+		try{
+			reader = new BufferedReader(new InputStreamReader(new TailInputStream(logFile, MAX_LENGTH), "UTF-8")); //$NON-NLS-1$
+			while (true) {
+				String line0 = reader.readLine();
+				if (line0 == null)
+					break;
+				String line = line0.trim();
+				
+				scanLine(line);
+			}
+		} catch (FileNotFoundException e) { // do nothing
+		} catch (IOException e) { // do nothing
```

Comment:
Don't forget we are reading Error Log file. I am not sure if we cannot read file we could write in it